src/Http/Event/UpdateStatusRequestHandler.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Http\Event;
+
+use Broadway\CommandHandling\CommandBusInterface;
+use CultuurNet\UDB3\Event\Commands\Status\UpdateStatus;
+use CultuurNet\UDB3\Event\ValueObjects\Status;
+use CultuurNet\UDB3\Event\ValueObjects\StatusReason;
+use CultuurNet\UDB3\Event\ValueObjects\StatusType;
+use CultuurNet\UDB3\Http\HttpFoundation\NoContent;
+use CultuurNet\UDB3\Http\Offer\UpdateStatusValidator;
+use CultuurNet\UDB3\Language;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class UpdateStatusRequestHandler
```

Comment:
Not sure if it would make sense to create an abstract class that contains the main logic and has one unimplemented method `CreateUpdateStatus` with a dedicated implementation for Events and one for Places.

Reply:
Additionally we could even change the `UpdateStatus` command so we only have one concrete command class, inside the `Offer` namespace, and then use the same trick with the combined repository as we did for the projectors? So 1 request handler, 1 command, 1 command handler, and 1 repository that tries to load the event and if it doesn't find it loads the place instead? And then it calls the `updateStatus` method on that aggregate.

Reply:
I'm not sure what part of the trade-off is better, but my reasoning here was "when in doubt, make it explicit".

Reply:
What's the difference with the projectors where you suggested to have one projector using a smart repository that loads either an event or a place? I'm suggesting basically the same but for the write side. 😄

Reply:
I care a lot less about model explicitness in projections than in write models. For me, projections are volatile by design and can easily be changed at will and replayed. I also merge the other PR by accident cause I thought the link to the remark you had was about the superfluous test scenario.

I'm okay with refactoring this but let's do it in a separate PR so we can easily evaluate the change at the end?

Reply:
It's just a suggestion that popped up in my head based on the solution for projectors. The current approach is also fine by me. I'm interested in exploring options to improve the general structure of our code like we did with the projectors, but it doesn't have to happen now if you believe this is the best for now for the write side.

Doing it in a separate PR is also fine, but only if you think it's worth the effort to try out and evaluate.

The things I care most about are the API endpoints/payloads/responses and event types/payloads in the event store, since those two are the hardest to refactor usually. (Because of backwards compatibility in the one case and persisted data in the other.) Aside from that everything is refactor-able so most of the times my comments with suggestions are just that, suggestions 😉 If something is a hard blocker for me I'll try to better make that clear. This is definitely not a hard blocker for me.

Reply:
To loop back to @LucWollants 's suggestion though: I'd go for either 100% explicitness like we have now, or a single request handler/command/command handler. Not an in-between solution with an abstract request handler which has an abstract `CreateUpdateStatus` method. Again just my $0.02 though.

Reply:
All good, let's do it in a separate PR